pkg/controller/jobframework/defaults.go

```diff
@@ -54,3 +55,17 @@ func ApplyDefaultForSuspend(ctx context.Context, job GenericJob, k8sClient clien
 	}
 	return nil
 }
+
+func ApplyDefaultLocalQueue(jobObj client.Object, defaultQueueExist bool) {
+	if !features.Enabled(features.LocalQueueDefaulting) || !defaultQueueExist {
```

Comment:
```suggestion
func ApplyDefaultLocalQueue(jobObj client.Object, defaultQueueExist func(ns string) bool) {
	if !features.Enabled(features.LocalQueueDefaulting) || !defaultQueueExist(jobObj.GetNamespace()) {
```

Reply:
We already send jobObj to ApplyDefaultLocalQueue, so we can retrieve it here to avoid duplicating logic across all webhooks.

Reply:
Updated